userdetails-cognito/src/main/groovy/au/org/ala/userdetails/CognitoUserService.groovy

```diff
@@ -667,4 +738,65 @@ class CognitoUserService implements IUserService {
         return code >= 200 && code < 300
     }
 
+    @Override
+    UserRecord findByUserNameOrEmail(String userName) {
+        return listUsers(userName, null, 1)?[0]
+    }
+
+    def getUserDetailsFromIdList(List idList){
+
+        List<UserType> users = []
+
+        ListUsersRequest request = new ListUsersRequest()
+                .withUserPoolId(poolId)
+                .withLimit(1)
+
+        idList.forEach{
+            if(it instanceof Number) {
+                request.withFilter("name = \"${it.toString()}\"")
+            }
+            else{
+                request.withFilter("username = \"${it.toString()}\"")
+            }
+            def response = cognitoIdp.listUsers(request)
+            users.addAll(response.users)
+        }
+
+        return users.stream().map { userType ->
+            cognitoUserTypeToUserRecord(userType, true)
+        }.toList()
+    }
+
+    def addCustomUserProperty(UserRecord user, String name, String value){
+        Collection<AttributeType> userAttributes = new ArrayList<>()
+
+        userAttributes.add(new AttributeType().withName('custom:' + name).withValue(value))
+
+        AdminUpdateUserAttributesRequest updateUserRequest =
+                new AdminUpdateUserAttributesRequest()
+                        .withUserPoolId(poolId)
+                        .withUsername(user.userName)
+                        .withUserAttributes(userAttributes)
+
+        return cognitoIdp.adminUpdateUserAttributes(updateUserRequest)
+    }
+
+    private void streamUserResults(ResultStreamer resultStreamer, List<UserRecord> results) {
+        resultStreamer.init()
+        try {
+            int count = 0
```

Comment:
Unused and can be removed - probably don't need this method if you're paging the results

Reply:
Fixed

Reply:
Shall I address this is in the clean up branch since there are few new changes here in clean up branch?

Reply:
Sounds good